fleetctl/fleetctl.go

```diff
@@ -845,3 +845,28 @@ func suToGlobal(su schema.Unit) bool {
 	}
 	return u.IsGlobal()
 }
+
+// checkUnitNames sanity checks unit names
+func checkUnitNames(names []string, allowTemplate bool) error {
+	units := make([]string, 0, len(names))
```

Comment:
sounds good. my original intention was to drop unit names that failed validation from the slice, but i feel like it's better to abort if any unit name fails checks.

unfortunately the combination of `unit.NewUnitNameInfo` and `unitNameMangle` is not terribly helpful, becuase `unit.NewUnitNameInfo` only returns nil if the unit is missing a `.` and `unitNameMangle` is appending `.service` for us. more checks are probably needed in `unit.NewUnitNameInfo`.